scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,52 @@
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str
+
+class BaseRobotsTxtParser():
+    def __init__(self, robotstxt_url):
+        raise NotImplementedError
+
+    def parse(self, robotstxt_body):
```

Comment:
I assume that the download of the `robots.txt` file will be done by Scrapy, as I assume is currently the case, and not the parser. Based on that assumption (is it wrong?), I wonder if it may make sense to merge the `parse` method into the `__init__` constructor.

Also, since classes implementing this API are expected to be `robot.txt` parsers, I wonder if we should rename `robotstxt_body` to simply `body`, `content` or similar.

Finally, documentation-wise, I think that it should be clear what the data type of the input parameter is. It could be a string, bytes, a file-like object…

Reply:
Fetching the `robots.txt` file will be done by Scrapy. I am combining the `parse` and `__init__` method.